tokio/src/time/driver/sleep.rs

```diff
@@ -182,20 +186,67 @@ pin_project! {
     #[derive(Debug)]
     #[must_use = "futures do nothing unless you `.await` or poll them"]
     pub struct Sleep {
-        deadline: Instant,
+        inner: Inner,
 
         // The link between the `Sleep` instance and the timer that drives it.
         #[pin]
         entry: TimerEntry,
     }
 }
 
+cfg_trace! {
+    #[derive(Debug)]
+    struct Inner {
+        deadline: Instant,
+        resource_span: tracing::Span,
+        async_op_span: tracing::Span,
+        time_source: ClockTime,
```

Comment:
nit/tioli: it occurs to me that we don't _really_ need to clone the time source into the `Inner`, as the `TimerEntry` struct already owns a `Handle` and exposes access to it via this accessor:
https://github.com/tokio-rs/tokio/blob/f3ed064a269fd72711e40121dad1a9fd9f16bdc0/tokio/src/time/driver/entry.rs#L401-L406

so, rather than cloning the `ClockTime` into the struct, we could just call `self.entry.handle().time_source()` whenever we need to get the time.  this makes the `Inner` struct one `Instant` smaller (since a `ClockTime` is a zero-sized `Clock` struct plus an `Instant`). OTTOH, this requires fewer pointer dereferences than accessing the time source through the handle every time. So, the current approach may actually be better. Anyway, just a thought.

Reply:
But that accessor is on `TimerShared` not `TimerEntry`, which is what we have in the struct